torch/_subclasses/meta_utils.py

```diff
@@ -859,6 +897,27 @@ def meta_tensor(
         )
         self.arg_cnt += 1
 
+        def metafy_fn(t: MetaTensorDesc, src: Source) -> torch.Tensor:
```

Comment:
I remember us talking about this `metafy_fn` before but I forget where we landed on it. IIRC, there is effectively an invariant today where `t` here is *guaranteed* to be a descriptor for a `CachedTensor` subclass (because `nested_int` always holds a CachedTensor` today. is that right? And if so we can we add some assertions to make that clear?

Reply:
Yeah that is right. Adding assertions to make it clear we're dealing with CachedTensor sounds good